src/internal/components/InternalThemeEditor.tsx

```diff
@@ -51,27 +51,40 @@ export const InternalThemeEditor = ({
     publishThemeConfiguration({
       payload: {
         saveThemeData: JSON.stringify(
-          themeHistory?.histories?.[themeHistory.index ?? 0]?.state?.data
+          themeHistories?.histories?.[themeHistories.index]?.data
         ),
       },
     });
+
+    clearThemeHistory();
+
+    setThemeHistories({
+      histories: [
+        {
+          id: themeHistories?.histories[themeHistories.index]?.id ?? "",
+          data: themeHistories?.histories?.[themeHistories.index]?.data ?? {},
```

Comment:
Turn this into an object that can be reused here and on line 54.
```
const currentThemeHistory = themeHistories?.histories[themeHistories.index];
```

I'm not sure what to do if currentThemeHistory is undefined, but can that ever actually happen?

Reply:
Hmmm technically it seems like no? Since we require themeHistoryFetched to be true before getting to this page.